### PR TITLE
[REF] CRM_Utils_Array::value() -> empty()

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Replaces `CRM_Utils_Array::value()` with `empty()` when the value is immediately cast to boolean.
The two statements have identical meaning but the latter is faster & easier to read.

Technical Details
----------------
Before/after should be functionally identical.

Comments
----------------------------------------
This was done via regex with some spot-checking.

### CODE REVIEW

File: CRM/Core/Form/RecurringEntity.php [hunk 1]
```diff
@@ -400,7 +400,7 @@ public static function postProcess($params = [], $type, $linkedEntities = []) {
         //Delete relations if any from recurring entity tables before inserting new relations for this entity id
         if ($params['entity_id']) {
           //If entity has any pre delete function, consider that first
-          if (CRM_Utils_Array::value('pre_delete_func', CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]) &&
+          if (!empty(CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['pre_delete_func']) &&
```
Comment (demeritcowboy) [inline CRM/Core/Form/RecurringEntity.php]: There's another instance in the same `if` statement that could also be converted.

File: CRM/Core/Form/RecurringEntity.php [hunk 2]
```diff
@@ -409,7 +409,7 @@ public static function postProcess($params = [], $type, $linkedEntities = []) {
             }
           }
           //Ready to execute delete on entities if it has delete function set
-          if (CRM_Utils_Array::value('delete_func', CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]) &&
+          if (!empty(CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['delete_func']) &&
             CRM_Utils_Array::value('helper_class', CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']])
```
Comment (demeritcowboy) [inline CRM/Core/Form/RecurringEntity.php]: Ditto - there's another instance in the same `if` statement that could also be converted.

File: CRM/Contact/Form/Search.php [hunk 3]
```diff
@@ -546,7 +546,7 @@ public function preProcess() {
 
     // assign context to drive the template display, make sure context is valid
     $this->_context = CRM_Utils_Request::retrieve('context', 'Alphanumeric', $this, FALSE, 'search');
-    if (!CRM_Utils_Array::value($this->_context, self::validContext())) {
+    if (!array_key_exists($this->_context, self::validContext())) {
```
Comment (demeritcowboy) [inline CRM/Contact/Form/Search.php]: Theoretically these aren't equal. I don't see any code outside this class that manipulates the $_validContext property directly, and it seems unlikely anyone would do something screwy with it like change it to null, but it's a public property. Seems low risk though.

File: CRM/Member/Import/Parser/Membership.php [hunk 4]
```diff
@@ -670,7 +670,7 @@ public function membership_format_params($params, &$values, $create = FALSE) {
           break;
 
         case 'membership_type_id':
-          if (!CRM_Utils_Array::value($value, CRM_Member_PseudoConstant::membershipType())) {
+          if (empty(CRM_Member_PseudoConstant::membershipType([$value]))) {
```
Comment (demeritcowboy) [inline CRM/Member/Import/Parser/Membership.php]: This seems wrong. Test failure too.

Comment (colemanw) [reply CRM/Member/Import/Parser/Membership.php]: Fixed using `array_key_exists()` which is what it was doing.

File: CRM/Contact/Form/Search.php (hunk 3 above)
Comment (colemanw) [reply CRM/Contact/Form/Search.php]: My thoughts too.

File: CRM/Core/Form/RecurringEntity.php (hunk 2 above)
Comment (colemanw) [reply CRM/Core/Form/RecurringEntity.php]: Fixed.

File: CRM/Core/Form/RecurringEntity.php (hunk 1 above)
Comment (colemanw) [reply CRM/Core/Form/RecurringEntity.php]: Fixed

**Comment by demeritcowboy** [comment]
Ok looks good I don't have anything more to add.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
test this please

**Comment by colemanw** [comment]
Unrelated fail

**Pull request merged**